File: apple/Sources/ReactNativeWatchOS/Modifiers/PreferredColorScheme.swift
import SwiftUI

/// SwiftUI `.preferredColorScheme(_:)`. Sets the preferred `ColorScheme` for
/// the enclosing presentation. A `nil`/absent `value` clears the preference
/// so the presentation follows the system appearance.
enum RNWPreferredColorSchemeModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("preferredColorScheme") { view, params, _ in
            // `parse` returns nil for a missing/null value, which maps to the
            // `ColorScheme?` "no preference" case SwiftUI expects.
            AnyView(view.preferredColorScheme(RNWColorSchemeModifier.parse(params.string("value"))))
        }
    }
}

File: src/modifiers/preferredColorScheme.ts
import { createModifier } from './createModifier';
import type { ColorScheme } from './colorScheme';

/// SwiftUI `.preferredColorScheme(_:)`. Sets the preferred color scheme for
/// the enclosing presentation (e.g. a sheet or the watch root). Pass `null`
/// (or omit) to clear the preference and follow the system appearance.
export function preferredColorScheme(value: ColorScheme | null = null) {
  return createModifier('preferredColorScheme', { value });
}
